module/motion/walk/TorsoMovement/src/TorsoMovement.cpp

```diff
@@ -0,0 +1,136 @@
+        using message::motion::TorsoTarget;
+        using utility::behaviour::RegisterAction;
+        using utility::input::LimbID;
+        using utility::input::ServoID;
+        using utility::math::matrix::Transform3D;
+        using utility::motion::kinematics::calculateLegJoints;
+
+
+        TorsoMovement::TorsoMovement(std::unique_ptr<NUClear::Environment> environment)
+            : Reactor(std::move(environment)), subsumptionId(size_t(this) * size_t(this) - size_t(this)) {
+
+            // Left for future use
+            // on<Configuration>("TorsoMovement.yaml").then([this](const Configuration& config) {
+
+            // });
+
+
+            on<Trigger<Sensors>, With<KinematicsModel>, With<TorsoTarget>>().then(
+                [this](const Sensors& sensors, const KinematicsModel& model, const TorsoTarget& target) {
+                    // Get support foot coordinate system
+
+                    Eigen::Affine3d Htf;  // support foot
+
+                    // Right foot is the support foot
+                    if (target.isRightFootSupport) {
+                        // Transform of right foot to torso
+                        Htf = Eigen::Affine3d(sensors.forwardKinematics[ServoID::R_ANKLE_ROLL]);
+                    }
+                    // Left foot is the support foot
+                    else {
+                        // Transform of left foot to torso
+                        Htf = Eigen::Affine3d(sensors.forwardKinematics[ServoID::L_ANKLE_ROLL]);
+                    }
+
+                    // Target given in support foot space
+                    Eigen::Affine3d Haf;
+                    Haf = target.Haf;
+
+                    // Create vector to target from support foot
+                    Eigen::Vector3d rAFf = -Haf.translation();
+                    // Position of target in torso space
+                    Eigen::Vector3d rATt = Htf * rAFf;
+                    rATt.normalize();
+
+                    // Find scale to reach target at specified time
+                    std::chrono::duration<double> time_left = target.timestamp - NUClear::clock::now();
+                    double distance = std::sqrt(std::pow(rATt.x(), 2) + std::pow(rATt.y(), 2) + std::pow(rATt.z(), 2));
+                    double scale;
+                    // 0.01 refers to 10 milliseconds as below
+                    if (time_left <= std::chrono::duration<double>::zero()) {  // Time has elapsed
+                        scale = 0.01;
+                    }
+                    else {  // There is time left to complete
+                        scale = (distance / time_left.count()) * 0.01;
+                    }
+
+                    // Create next torso target in torso space
+                    Eigen::Vector3d rT_tTt = rATt * scale;
+                    // Create vector from torso to foot target
+                    Eigen::Vector3d rF_tFt = -rT_tTt;
+                    Eigen::Vector3d rF_tTt = Htf * rF_tFt;
```

Comment:
If this is the only place that `rF_tFt` is used then just change this `Eigen::Vector3d rF_tTt = Htf * (-rT_tTt);` and remove the line above (`Eigen::Vector3d rF_tFt = -rT_tTt;`)